src/effects/PitchEffect.js

```diff
@@ -8,32 +8,38 @@ var Tone = require('tone');
 
 function PitchEffect () {
     this.value = 0;
+    this.ratio = 1;
 
     this.tone = new Tone();
 }
 
 PitchEffect.prototype.set = function (val, players) {
     this.value = val;
+    this.ratio = this.getRatio(this.value);
     this.updatePlayers(players);
 };
 
 PitchEffect.prototype.changeBy = function (val, players) {
     this.set(this.value + val, players);
 };
 
-PitchEffect.prototype.getRatio = function () {
-    return this.tone.intervalToFrequencyRatio(this.value / 10);
+PitchEffect.prototype.getRatio = function (val) {
+    return this.tone.intervalToFrequencyRatio(val / 10);
+};
+
+PitchEffect.prototype.updatePlayer = function (player) {
+    player.setPlaybackRate(this.ratio);
 };
 
 PitchEffect.prototype.updatePlayers = function (players) {
     if (!players) return;
 
-    var ratio = this.getRatio();
-    for (var i=0; i<players.length; i++) {
-        players[i].setPlaybackRate(ratio);
+    for (var md5 in players) {
+        this.updatePlayer(players[md5]);
```

Comment:
I'd actually probably iterate through the keys, if I were writing it:

```js
for (let md5 of Object.keys(players)) {
  this.updatePlayer(players[md5]);
}
```

..Though, now that I look at it, you're really just iterating through the object's values, so you could do this instead:

```js
for (let player of Object.values(players)) {
  this.updatePlayer(player)
}
```

If you're limited to avoiding for..of loops (are you?) you *would* probably have to go with your code, though, @cwillisf.